src/Concerns/UsesPresenters.php

```diff
@@ -9,18 +9,62 @@
  */
 trait UsesPresenters
 {
+    /*
+    |--------------------------------------------------------------------------
+    | Presenters
+    |--------------------------------------------------------------------------
+    |
+    | Presenters can be defined as a single item in $this->presenter or
+    | as an array of $this->presenters. If neither are set, then we will
+    | generate a default based on the Model and type (if set).
+    |
+    */
+
     /**
-     * Check the given presenters value exists or not
-     *
-     * @param string $type
-     * @return mixed
+     * @throws PresenterException
      */
-    public function present(string $type = 'default')
+    public function present(string $type = 'default'): mixed
     {
-        if (array_key_exists($type, $this->presenters)) {
-            return new $this->presenters[$type]($this);
+        $presenter = $this->getDefaultPresenterName($type);
+
```

Comment:
I see a __Code smell__ in here, consider using aggressive condition statement, to clean the code, and understandable.

```suggestion
if (
    is_array($this->presenters) &&
    array_key_exists($type, $this->presenters ?? [])
) {
  $presenter =$this->presetners[$type];
}

if (is_string($this->presenter ?? null)) {
  $presenter = $this->presenter;
}

if(class_exists($presenter)) {
   return new $presenter($this);
}

throw new PresenterException();
```